### DISCUSSION TITLE
Multiple Issues with Ankimon Add-on on macOS (M1 Chip)

### DISCUSSION THREAD

**Discussion by Ankiholic** [Unlucky-Life/ankimon#147] [Ideas]
I have encountered several issues while using the Ankimon add-on on my macOS device with an M1 chip. Below are the details of the problems and suggested fixes:

1. "About Anki" Menu Item Issue:
- When accessing the "About Anki" menu item in the main ribbon, it displays the Ankimon "About" page instead of the Anki "About" page.
- This was resolved by adding 'test_action13.setMenuRole(QAction.MenuRole.NoRole)' in the code. However, this fix should be implemented by default.

2. Main Screen Issues:
- The enemy Pokémon is visible in the bottom left, and its HP is shown in the bottom right. However, my own Pokémon is not displayed on the screen.
- 
<img width="746" alt="image" src="https://github.com/Unlucky-Life/ankimon/assets/173546808/028643c2-8723-43cd-91f6-8031c773fe22">

- The "Open Ankimon Window" option is available in the menu, but it would be beneficial if the player's Pokémon were displayed by default on each flash card like the enemy is described above.

3. Battle Mechanics Issues:
- The enemy Pokémon does not take any action during its turn. After my turn, I will complete the next flash card (good/again etc), but the enemy Pokémon does nothing.
- As a result, the battles are one-sided, with the player always winning since the enemy Pokémon does not attack.

4. Item and Experience Issues:
- No items, such as Elixirs, appear during or after the battles, and there is no mechanism for using or earning items.

5. Pokémon Collection Issue:
- In the "Show Pokémon Collection" tab, the "Pick as Main Pokémon" option can be repeatedly clicked to refresh the current fight. - This allows the player to cycle through fights until they encounter a desired Pokémon, which undermines the game's challenge.

General Observations:
- I have followed the warning about not downloading the resources whilst in full-screen mode and ensuring the laptop does not hibernate or activate the screensaver. Despite these precautions, the issues persist.
- I am using the most updated version of Anki (Version ⁨24.06.2 (33a92379)⁩), and these problems continue even after trying to entirely reinstall add-on.

Thank you for your work on this add-on. I hope this feedback helps improve Ankimon's functionality and user experience. I'm so keen to use this!

**Comment by Ankiholic** [comment]
**Follow-Up: Additional Observations and Suggestions**

**Enemy Attacks:**
- There have been sporadic instances where the enemy Pokémon has attacked my Pokémon, causing HP to decrease. However, I haven't seen (i guess i could have missed it though) text indicating the enemy’s attack (e.g., "Enemy Pokémon used Burn for 2 damage").
- This occurred very infrequently, about 1-2 instances over approximately 50 cards. Attempts to replicate this by spamming cards (either all "Again," all "Good," or all "Easy") were unsuccessful.

**Suggestion:**
Since the combat is turn-based, each flashcard should correspond to either the player's move or the enemy's move. The effectiveness of the move could be based on the card’s response:
  - Player’s turn: One flashcard represents the player’s move.
  - Enemy’s turn: The next flashcard represents the enemy’s move.
  - Card performance multipliers:
    - Player’s turn: Correct card = 1x multiplier on the player’s attack; incorrect card = 0.25x or 0.5x multiplier.
    - Enemy’s turn: Correct card = 1x multiplier on the enemy’s attack; incorrect card = 1.5x or 2x multiplier.

**Item Drops:**
- I have sporadically received items, with only three items being dropped over an extended period. Not sure if this is standard drop rate or not.

**Feature Request: Switch Pokémon Mid-Fight**

I have not found an option to manually switch Pokémon mid-fight. Could this feature be added to Ankimon for a more authentic experience?

**Comment by Unlucky-Life** [comment]
@Ankiholic 
1. Yes, this is true (it was a mac specific issue, and im using a windows, therefore i didnt have this issue before)
2. You can change the display of the pokemons in the reviewer with 'show_mainpkmn_in_reviewer' setting in the addons settings: 
![image](https://github.com/Unlucky-Life/ankimon/assets/77027147/2610b100-b9e9-439e-bf7f-c3415c421a3e)

You can also open and close the ankimon window by using the "ankimon key" you can choose in the settings options.

3. You can also let the enemy pokemon attack you by changing 'dmg_in_reviewer' setting to true.

4. You earn items randomly - at every anki session start up there is a random value picked of cards that need to be reviewed. If you have reviewed them, a pop up will appear letting you know that you have received an item.

Right now Items that are working:
- Healing Items
- Fossil Items => You can evolve the pokemon from them
- Evolution Trigger Items -> (here is a bug that if multiple pokemon can be possibly evolved, that the system wont registrate it)


5. That is true, currently reworking the Switch Main Pokemon possibility (because there was an issue before when that was an option.)

**Reply by Ankiholic** [reply]
Thank you so much for your prompt response and assistance. I truly appreciate the time and effort you’ve put into the Ankimon add-on.

I must apologise for missing the extensive configuration options available within the add-on settings. These settings allowed for quick changes, such as enabling damage and displaying my Pokémon in the reviewer. Your guidance on this was incredibly helpful.

Thank you again for considering my feedback. Your work on this add-on has been fantastic. It’s now working brilliantly, and it has greatly enhanced my Anki experience. I am very grateful for your continued support and dedication.

**Comment by Unlucky-Life** [comment]
@Ankiholic 
I think the suggestion with the change of round based battle mechanic can be tempting for users to not press "again" on their own turn, while enemys turn to press "again" more. That could be an issue.

But i think we could rethink that definetly.

Btw. You gain xp by defeating pokemon, catching those not allow you to gain xp.

**Comment by Unlucky-Life** [comment]
@Ankiholic If you have set your option to allow dmg in reviewer to true - currently after 2 cards you and the enemy pokemon will receive the chance to attack each other. while the enemy pokemon will only attack if you have a multiplier below 1 - (meaning you have pressed again or hard)

**Reply by Ankiholic** [reply]
Yeah i think I actually like the turn being every 2 cards and not every card as it means you don't progress through pokemon leveling etc too quickly, especially for those of us with many cards.

**Comment by Unlucky-Life** [comment]
@Ankiholic Thank you for the thorough message and bug testing ! Very helpful :)
There are also here on GitHub the latest release, with a few new features you can check out.
https://github.com/Unlucky-Life/ankimon/releases/tag/1.288
